lib/index.js

```diff
@@ -0,0 +1,255 @@
+  assert.string(kekId, 'options.kekId');
+  assert.string(wrappedKey, 'options.wrappedKey');
+
+  const {key: kek} = await _getKeyRecord({id: kekId, controller});
+  if(kek.algorithm !== 'AES-KW') {
+    throw new Error(`Unknown unwrapping algorithm "${kek.algorithm}".`);
+  }
+
+  const key = await _aesUnwrapKey({kek, wrappedKey});
+  return {key};
+};
+
+/**
+ * Signs some data. Note that the data will be sent to the server, so if
+ * this data is intended to be secret it should be hashed first. However,
+ * hashing the data first may present interoperability issues so choose
+ * wisely.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.controller - The ID of the controller of the key.
+ * @param {string} options.keyId - The ID of the signing key to use.
+ * @param {Uint8Array|string} options.data - The data to sign as a Uint8Array
+ *   or a base64url-encoded string.
+ *
+ * @returns {Promise<Object>} An object containing `{signature}`.
+ */
+exports.sign = async ({controller, keyId, data}) => {
+  assert.string(controller, 'options.controller');
+  assert.string(keyId, 'options.keyId');
+  assert.string(data, 'options.data');
+
+  const {key} = await _getKeyRecord({id: keyId, controller});
+
+  const signature = await _hs256Sign({key, data});
+  return {signature: base64url.encode(signature)};
+};
+
+/**
+ * Verifies some data. Note that the data will be sent to the server, so if
+ * this data is intended to be secret it should be hashed first. However,
+ * hashing the data first may present interoperability issues so choose
+ * wisely.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.controller - The ID of the controller of the key.
+ * @param {string} options.keyId - The ID of the signing key to use.
+ * @param {Uint8Array|string} options.data - The data to sign as a Uint8Array
+ *   or a base64url-encoded string.
+ * @param {string} options.signature - The base64url-encoded signature to
+ *   verify.
+ *
+ * @returns {Promise<Object>} An object containing `{verified}`.
+ */
+exports.verify = async ({controller, keyId, data, signature}) => {
+  assert.string(controller, 'options.controller');
+  assert.string(keyId, 'options.keyId');
+  assert.string(data, 'options.data');
+  assert.string(signature, 'options.signature');
+
+  const {key} = await _getKeyRecord({id: keyId, controller});
+
+  const verified = await _hs256Verify({key, data, signature});
+  return {verified};
+};
+
+/**
+ * Gets a previously stored key record.
+ *
+ * @ignore
+ * @param {Object} options - The options to use.
+ * @param {string} options.controller - The controller of the key.
+ * @param {string} options.id - The ID of the key.
+ *
+ * @returns {Promise<Object>} The key record.
+ */
+async function _getKeyRecord({controller, id}) {
+  assert.string(controller, 'options.controller');
+  assert.string(id, 'options.id');
+
+  const record = await database.collections.ssm.findOne(
```

Comment:
I see that `id` is a unique index, so controller is really superfluous here, is there some implicit contract in place here?  Should we eliminate `controller` or make the unique index be a compound index on `controller` + `id`

Reply:
Again, not sure what to do about this. I agree it's a problem we need to solve, but it's related to key IDs ... which we don't have a clear story around.

Reply:
There is no "proving" control over key IDs ... which is a problem if people "claim" key IDs that shouldn't belong to them.